File: src/apple/Packages/MapleUI/Sources/MapleUI/MoleculesL2/MuiPreviewList.swift
// MuiPreviewList.swift — Maple UI Molecules-L2 (unified-component-
// catalog.md §3). Before -> after row list, built from List Row, Text.

import SwiftUI

public struct MuiPreviewItem: Identifiable, Sendable {
    public let id: String
    public let before: String
    public let after: String

    public init(id: String, before: String, after: String) {
        self.id = id
        self.before = before
        self.after = after
    }
}

public struct MuiPreviewList: View {
    public let items: [MuiPreviewItem]
    public let pressed: ((String) -> Void)?

    public init(items: [MuiPreviewItem], pressed: ((String) -> Void)? = nil) {
        self.items = items
        self.pressed = pressed
    }

    public var body: some View {
        VStack(spacing: 0) {
            ForEach(items) { item in
                MuiListRow(label: item.before, pressed: { pressed?(item.id) }) {
                    MuiText(item.after, variant: .chipLabel, color: .muted, truncate: true)
                }
            }
        }
    }
}

#Preview("MuiPreviewList") {
    MuiPreviewList(items: [
        MuiPreviewItem(id: "1", before: "IMG_0042.dng", after: "iceland-glacier-01.dng"),
        MuiPreviewItem(id: "2", before: "IMG_0043.dng", after: "iceland-glacier-02.dng"),
    ])
    .background(MuiTokens.bg)
}

File: src/windows/Maple.WinUI/MapleUI/MoleculesL2/MuiPreviewList.cs
using System;
using System.Collections.Generic;
using Microsoft.UI.Xaml;
using Microsoft.UI.Xaml.Controls;
using Maple.UI.Atoms;

namespace Maple.UI
{
    /// <summary>One before/after row.</summary>
    public sealed record MuiPreviewItem(string Id, string Before, string After);

    /// <summary>
    /// Maple.UI Preview List molecule (unified-component-catalog.md §3,
    /// "Preview List" row: "Before → after row list", built from List Row,
    /// Text) — a stack of <see cref="MuiListRow"/>s (label = before value,
    /// trailing text = after value), the live-preview list every batch
    /// operation (Batch Rename's template preview, a Suggestion sweep)
    /// shows before it commits.
    /// </summary>
    public sealed class MuiPreviewList : ContentControl
    {
        public static readonly DependencyProperty ItemsProperty =
            DependencyProperty.Register(nameof(Items), typeof(IReadOnlyList<MuiPreviewItem>), typeof(MuiPreviewList),
                new PropertyMetadata(null, (d, _) => ((MuiPreviewList)d).Rebuild()));

        public IReadOnlyList<MuiPreviewItem>? Items
        {
            get => (IReadOnlyList<MuiPreviewItem>?)GetValue(ItemsProperty);
            set => SetValue(ItemsProperty, value);
        }

        public event EventHandler<string>? Pressed;

        private readonly StackPanel _root = new() { Orientation = Orientation.Vertical, Spacing = 0 };

        public MuiPreviewList()
        {
            Content = _root;
            IsTabStop = false;
            Rebuild();
        }

        private void Rebuild()
        {
            _root.Children.Clear();
            foreach (var item in Items ?? Array.Empty<MuiPreviewItem>())
            {
                var row = new MuiListRow
                {
                    Label = item.Before,
                    TrailingContent = new MuiText { Text = item.After, Variant = MuiTextVariant.ChipLabel, ColorRole = MuiTextColorRole.Muted, Truncate = true },
                };
                var id = item.Id;
                row.Pressed += (_, _) => Pressed?.Invoke(this, id);
                _root.Children.Add(row);
            }
        }
    }
}
